### PR TITLE
Add types for react-swipeable-bottom-sheet

### PR DESCRIPTION
**Pull request by yonatand**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/react-swipeable-bottom-sheet/package.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+    "private": true,
+    "name": "@types/react-swipeable-bottom-sheet",
+    "version": "1.1.9999",
+    "projects": [
+        "https://github.com/manufont/react-swipeable-bottom-sheet"
+    ],
+    "dependencies": {
+        "@types/react": "*",
```
Comment (jakebailey) [inline types/react-swipeable-bottom-sheet/package.json:9]: This package peer deps on React v16; does this actually work?

Comment (yonatand) [reply types/react-swipeable-bottom-sheet/package.json]: You're right, I tested it with React 18 and didn't notice the peer dependencie.
Changed the package.json to require React ^16 and change the type definition to work with that.

File: types/react-swipeable-bottom-sheet/package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
         "https://github.com/manufont/react-swipeable-bottom-sheet"
     ],
     "dependencies": {
-        "@types/react": "*",
+        "@types/react": "^16",
         "@types/react-swipeable-views": "*"
```
Comment (jakebailey) [inline types/react-swipeable-bottom-sheet/package.json:10]: Confusingly, this dep depends on `@types/react@*`...

**Review by jakebailey — APPROVED** [review]

**Comment by yonatand** [comment]
Ready to merge

**Pull request merged**